soup.go

```diff
@@ -15,20 +15,67 @@ import (
 	"golang.org/x/net/html"
 )
 
-// Root is a structure containing a pointer to an html node, the node value, and an error variable to return an error if occurred
+// ErrorType defines types of errors that are possible from soup
+type ErrorType int
+
+const (
+	// ErrUnableToParse will be returned when the HTML could not be parsed
+	ErrUnableToParse ErrorType = iota
+	// ErrElementNotFound will be returned when element was not found
+	ErrElementNotFound
+	// ErrNoNextSibling will be returned when no next sibling can be found
+	ErrNoNextSibling
+	// ErrNoPreviousSibling will be returned when no previous sibling can be found
+	ErrNoPreviousSibling
+	// ErrNoNextElementSibling will be returned when no next element sibling can be found
+	ErrNoNextElementSibling
+	// ErrNoPreviousElementSibling will be returned when no previous element sibling can be found
+	ErrNoPreviousElementSibling
+	// ErrCreatingGetRequest will be returned when the get request couldn't be created
+	ErrCreatingGetRequest
+	// ErrInGetRequest will be returned when there was an error during the get request
+	ErrInGetRequest
+	// ErrReadingResponse will be returned if there was an error reading the response to our get request
+	ErrReadingResponse
+)
+
+// Error allows easier introspection on the type of error returned.
+// If you know you have a Error, you can compare the Type to one of the exported types
+// from this package to see what kind of error it is, then further inspect the Error() method
+// to see if it has more specific details for you, like in the case of a ErrElementNotFound
+// type of error.
+type Error struct {
```

Comment:
I did have this as SoupError initially, but my editor reminded me that this type will be used as `soup.SoupError` which means it will stutter. `soup.Error` looks decent, like in the example code.

Reply:
Yes let's keep it as `Error` alone